kythe/typescript/indexer.ts

```diff
@@ -416,6 +416,13 @@ class Visitor {
       const kType = this.getSymbolName(sym, TSNamespace.TYPE);
       this.emitNode(kType, 'absvar');
       this.emitEdge(this.newAnchor(param.name), 'defines/binding', kType);
+      // ...<T extends A>
+      if (param.constraint) {
+        const superType = this.visitType(param.constraint);
```

Comment:
Does this handle things like "<T, K extends keyof T>" ?

(and is there only "extends" in TS, no "super" like Java has?)

And are there recursive types, e.g. <T extends Foo<T>>, and if so does this handle those?

Reply:
#1: yes, added a test
#2: right
#3: yes, added a test